src/to_ir.rs

```diff
@@ -16,15 +17,15 @@ use std::rc::Rc;
 /***********************************/
 
 pub struct PropertyAndUdfVisitor {
-    properties: Vec<Property>,
-    udf_calls: Vec<UdfCall>,
+    properties: IndexSet<Property>,
```

Comment:
What was the impetus to switch to IndexSets?

Reply:
We only want unique properties and calls. Unless something changes in the way UDFs are excecuted and properties are fetched.